File: python/examples-pytest/steps/tables_and_docstrings.steps.py
from var import define_state

stimulus, sensor = define_state(lambda: {})


# Whole-table mode: the table arrives as a list of rows (header row first).
# It is this sensor's only slot, so return the reproduced table bare — Vár
# compares every cell.
@sensor("Uppercase each one:")
def _(state, rows):
    return [{"before": before, "after": before.upper()} for before, *_ in rows[1:]]


# Doc-string mode: two slots ({word} plus the trailing doc string), so return
# one element per slot.
@sensor("Greet {word}:")
def _(state, name, doc):
    return [name, f"Hello, {name}!\n"]

File: examples/ruby-rspec/steps/tables_and_docstrings.steps.rb
require "oselvar/var"

param, stimulus, sensor = steps

# Whole-table mode: the table arrives as an array of rows (header row first).
# It is this sensor's only slot, so return the reproduced table bare — Vár
# compares every cell.
sensor.("Uppercase each one:") do |_state, rows|
  rows[1..].map { |before, *| { "before" => before, "after" => before.upcase } }
end

# Doc-string mode: two slots ({word} plus the trailing doc string), so return
# one element per slot.
sensor.("Greet {word}:") do |_state, name, _doc|
  [name, "Hello, #{name}!\n"]
end
